packages/cli/src/cmds/beacon/handler.ts

```diff
@@ -33,6 +34,7 @@ const HOURS_TO_MS = 3600 * 1000;
  * Runs a beacon node.
  */
 export async function beaconHandler(args: BeaconArgs & GlobalArgs): Promise<void> {
+  process.env.UV_THREADPOOL_SIZE = os.availableParallelism().toString();
```

Comment:
This is such a critical line of code, might wanna extract this in a separate function with a few notes why we need to increase default UV thread pool size (or add a comment here).

I feel like it's also important to hightlight that this must be done before any async operation is done that involves the libuv threadfile, e.g. like reading a file.

or is there some mechanism to throw an error if the UV threadpool can no longer be resized?

Eg. we do a file read here, but it's sync call so I think it shouldn't be an issue
https://github.com/ChainSafe/lodestar/blob/2dae605a9c83cfc349d9c89e85d6463006cf1c38/packages/cli/src/applyPreset.ts#L67

Reply:
Good point!!  Updated.

> or is there some mechanism to throw an error if the UV threadpool can no longer be resized?

Not to my knowledge and ive dug pretty extensively.  There is no way to tell if the threadpool has been initialized or not.  Nor is there any way to set this value aside from the ENV.

I refactored this out into a function and added some notes so its clear what happening.